### ISSUE TITLE
Space Bar Behaves Differently in Combat Compared to 13066

### ISSUE THREAD

**Issue by beelee1** [triplea-game/triplea#6132]
**How can the problem be recreated?**
Have a battle with multiple casualty options for attacker. When hitting space in 13066 it will ask which unit you want to select. In Pre it just powers through using the default. A quick flash is all I can see. Both releases combat settings are the default/same.

**When the problem is fixed, what should have happened instead?**
Well Idk if it's broken or deliberately changed but I like the way 13066 does it. I tried using cursor but then it just goes unhighlighted and i have to hit space. It's also inconsistent. Sometimes it won't do it and works same as 13066. I can't figure out why. Not doing anything different that I'm aware of.
**Which Engine Version are you using?** 
18587

**Screenshots** 

**Zipped Save Game**
[test combat pre 18587.zip](https://github.com/triplea-game/triplea/files/4432228/test.combat.pre.18587.zip)

**Comment by beelee1** [comment]
Here's the same situation with 13066. Just hit space during combat and you'll see the difference.

[test combat 13066.zip](https://github.com/triplea-game/triplea/files/4432242/test.combat.13066.zip)

**Comment by beelee1** [comment]
here's another example with 18613

[18613 combat window space bar freeze.zip](https://github.com/triplea-game/triplea/files/4434217/18613.combat.window.space.bar.freeze.zip)
 just hit done and do the combat. Space doesn't do anything but when you drag the window out of the way you see the appropriate one.

I won't post any more examples unless requested as it's most likely the same thing.

**Comment by DanVanAtta** [comment]
@beelee1 i suspect you're being hit with dialogs appearing behind the combat window. Then on a space bar click, the click is going to the hidden dialog and hence 'powers through'.

I just tried out the two saves you had (thank you!) and did not see any difference. Though, I also saw all dialogs coming to the front as well.

**Comment by DanVanAtta** [comment]
@beelee1 , you see this problem every time, or some of the time?

**Comment by DanVanAtta** [comment]
- I think this problem relates to battle pop-up windows going behind the battle window
- we had a problem when battle pop-up windows were displayed the battle window would go behind the main map. To fix this we added a (hack) 'alwaysOnTop(true)' to BattleDisplay (instantiated from BattlePanel).

One key observation, when you minimize the battle window, the pop-up is not minimized. If you minimize the map, the battle pop-up (like select casualties) is minimized. This seems to be the key. If the pop-up became associated with the main map, then when the pop-up is displayed the map is brought to the front and that would hide the battle. Setting the battle to be 'always on top' then "fixed" this, but we can still see if you minimize the respective windows you can see how they are associated.

'Always on top' is a good thing for battle window, but we want the pop-up to be associated with the battle window, not the main map. That means when the battle window is minimized and then un-minimized that the pop-up (like submerge subs) should go with it.

'Always on top' behavior is OS and JVM specific, that explains why this problem is sporadic.

I was able to bisect this problem down to these two commits:

12362cf5f < not okay
a3818cd7b < okay

1236cf is this PR: https://github.com/triplea-game/triplea/pull/4899/files

In theory something changed in that PR that caused this behavior. If we can find and fix/undo that update to get the right association of the battle window pop-up back to the battle window, we should have this properly fixed.

**Comment by DanVanAtta** [comment]
I think I may have spotted it :smile: 

Going to eat dinner, PR coming after.

**Comment by DanVanAtta** [comment]
PR is up: https://github.com/triplea-game/triplea/pull/6227

Once that is merged, I'll be eager to know @beelee1 if you continue to see this problem (as it seems you can pretty consistently see this problem; on my OS+java I do not see it). The 'subs submerging' dialog appearing behind the battle window would be  the same issue.

**Comment by beelee1** [comment]
@DanVanAtta right on I'll be ready for some test mode soon. Having supper myself :) The above question is yes happens all the time.

**Comment by beelee1** [comment]
also fwiw I'm on linux mint. Idk if that matters. im gonna bag it and just go back to ubuntu soon

**Comment by DanVanAtta** [comment]
@beelee1 an update is built into: https://github.com/triplea-game/triplea/releases/tag/2.0.18786

Please give it a try and let us know  :crossed_fingers:

**Comment by beelee1** [comment]
Will do.  Fire it up in a bit. :) How do you keep all this stuff compartmentalized lol :)

**Comment by DanVanAtta** [comment]
@beelee1 any luck confirming if you still this problem or looks fixed?

**Comment by beelee1** [comment]
@DanVanAtta seems to be working correctly on 18830 pre

**Comment by DanVanAtta** [comment]
That is great news!
